File: K번째수.js
function solution(array, commands) {
    let answer = [];
    let cutted = []
    let from = 0;
    let to = 0;
    let nd = 0;

    for (let i = 0; i < commands.length; i++){
        cutted = [];
        
        from = commands [i][0];
        to = commands [i][1];
        nd = commands [i][2];

        if (to > 1){
            //인덱스 값으로 변환
            from = from - 1;
            
            //자르기
            cutted = array.slice(from, to);
            cutted = cutted.sort((a, b) => a-b);
            
            //넣기
            answer.push(cutted[nd-1]);
        }
        else if (to == 1){
            answer.push(array[0]);
        }
        else if (to == array.length){
            cutted = array.slice(from);
            cutted = cutted.sort((a, b) => a-b);
            
            //넣기
            answer.push(cutted[nd-1]);
        }
    }

    return answer;
}

solution([1, 5, 2, 6, 3, 7, 4], [[2, 5, 3], [4, 4, 1], [1, 7, 3]]);

File: K번째수.py
def solution(array, commands):
    answer = []
    
    for command in commands:
        i = command[0]
        j = command[1]
        k = command[2]
    
        ind = 0
        kList = []

        for l in array:
            if i-1 <= ind <= j-1:
                kList.append(l)
            ind += 1
        kList = sorted(kList)
        answer.append(kList[k-1])
    
    return answer